src/main/groovy/net/vivin/gradle/versioning/VersionUtils.groovy

```diff
@@ -117,6 +113,14 @@ class VersionUtils {
                 return headTag - TAG_PREFIX_PATTERN
             }
         }
+
+        if(version.snapshot) {
+            result = "$result-$version.config.snapshotSuffix"
+        } else if (versionByTag.containsValue(result)) {
+            throw new BuildException("Determined version '$result' already exists in the repository at '$repository.directory'.\nFix your bumping or manually create a tag with the intended version on the commit to be released.", null)
```

Comment:
Maybe change to "A tag that identifies version [...]"?

Also should we warn them about manually creating a tag? Because that would amount to overriding the previous version, right? Or at the very least, make that version ambiguous.